nixos/modules/installer/tools/tools.nix

```diff
@@ -104,145 +213,48 @@ in
-        #   packages = with pkgs; [
-        #     firefox
-        #     tree
-        #   ];
-        # };
-
-        # List packages installed in system profile. To search, run:
-        # \$ nix search wget
-        # environment.systemPackages = with pkgs; [
-        #   vim # Do not forget to add an editor to edit configuration.nix! The Nano editor is also installed by default.
-        #   wget
-        # ];
-
-        # Some programs need SUID wrappers, can be configured further or are
-        # started in user sessions.
-        # programs.mtr.enable = true;
-        # programs.gnupg.agent = {
-        #   enable = true;
-        #   enableSSHSupport = true;
-        # };
-
-        # List services that you want to enable:
-
-        # Enable the OpenSSH daemon.
-        # services.openssh.enable = true;
-
-        # Open ports in the firewall.
-        # networking.firewall.allowedTCPPorts = [ ... ];
-        # networking.firewall.allowedUDPPorts = [ ... ];
-        # Or disable the firewall altogether.
-        # networking.firewall.enable = false;
-
-        # Copy the NixOS configuration file and link it from the resulting system
-        # (/run/current-system/configuration.nix). This is useful in case you
-        # accidentally delete configuration.nix.
-        # system.copySystemConfiguration = true;
-
-        # This option defines the first version of NixOS you have installed on this particular machine,
-        # and is used to maintain compatibility with application data (e.g. databases) created on older NixOS versions.
-        #
-        # Most users should NEVER change this value after the initial install, for any reason,
-        # even if you've upgraded your system to a new NixOS release.
-        #
-        # This value does NOT affect the Nixpkgs version your packages and OS are pulled from,
-        # so changing it will NOT upgrade your system - see https://nixos.org/manual/nixos/stable/#sec-upgrading for how
-        # to actually do that.
-        #
-        # This value being lower than the current NixOS release does NOT mean your system is
-        # out of date, out of support, or vulnerable.
-        #
-        # Do NOT change this value unless you have manually inspected all the changes it would make to your configuration,
-        # and migrated your data accordingly.
-        #
-        # For more information, see `man configuration.nix` or https://nixos.org/manual/nixos/stable/options#opt-system.stateVersion .
-        system.stateVersion = "${config.system.nixos.release}"; # Did you read the comment?
-
-      }
-    '';
-
-    environment.systemPackages =
-      [ nixos-build-vms
-        nixos-install
-        nixos-rebuild
-        nixos-generate-config
-        nixos-option
-        nixos-version
-        nixos-enter
-      ];
+  imports = let
+    mkToolModule = { name, package ? pkgs.${name} }: { config, ... }: {
+      options.system.tools.${name}.enable = lib.mkEnableOption "${name} script" // {
```

Comment:
Do you have a specific use-case for making all these tools toggle-able?

Reply:
I mostly want my server boxes to not have anything but nixos-version, but it's easier to make all of them optional than to bikeshed over which exact ones should be.